### PR TITLE
Changes for edge-19.12.3

### PR DESCRIPTION
**Pull request by cpretzer**
## edge-19.12.3

* CLI
  * Increased the comprehensiveness of `linkerd check --pre`
  * Added TLS certificate validation to `check` and `upgrade` commands
* Controller
  * Increased minimum kubernetes version to 1.13.0
  * Added support for pod ip and service cluster ip lookups in the destination 
    service
  * Added recommended kubernetes labels to control-plane
  * Added preStop hook into the proxy sidecar container to stop it last (a huge 
    commit from @KIVagant, thanks!)
  * Added a pre-sign check to te identity service
* Web UI
  * Added logic to pause dashboard polling activity when the page is not in 
    focus
* Proxy
  * Added a timeout to release resolutions to idle balancers
  * Reclassified gRPC status codes as non-errors
* Internal
  * Updated helm charts to follow best practices using proper casing (thanks 
    @Pothulapati!)

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,3 +1,26 @@
...
+* Controller
+  * Increased minimum kubernetes version to 1.13.0
+  * Added support for pod ip and service cluster ip lookups in the destination 
+    service
+  * Added recommended kubernetes labels to control-plane
+  * Added preStop hook into the proxy sidecar container to stop it last (a huge 
+    commit from @KIVagant, thanks!)
+  * Added a pre-sign check to te identity service
+* Web UI
+  * Added logic to pause dashboard polling activity when the page is not in
```
Comment (scottcarol) [inline CHANGES.md]: I would rephrase as: "Increased the speed of the dashboard by pausing network activity when the dashboard is not visible to the user"

File: CHANGES.md [hunk 2]
```diff
@@ -1,3 +1,26 @@
...
+  * Increased the comprehensiveness of `linkerd check --pre`
+  * Added TLS certificate validation to `check` and `upgrade` commands
+* Controller
+  * Increased minimum kubernetes version to 1.13.0
+  * Added support for pod ip and service cluster ip lookups in the destination 
+    service
+  * Added recommended kubernetes labels to control-plane
+  * Added preStop hook into the proxy sidecar container to stop it last (a huge 
+    commit from @KIVagant, thanks!)
+  * Added a pre-sign check to te identity service
```
Comment (scottcarol) [inline CHANGES.md]: `the` identity service

File: CHANGES.md [hunk 3]
```diff
@@ -1,3 +1,26 @@
...
+* CLI
+  * Increased the comprehensiveness of `linkerd check --pre`
+  * Added TLS certificate validation to `check` and `upgrade` commands
+* Controller
+  * Increased minimum kubernetes version to 1.13.0
+  * Added support for pod ip and service cluster ip lookups in the destination 
+    service
+  * Added recommended kubernetes labels to control-plane
+  * Added preStop hook into the proxy sidecar container to stop it last (a huge 
+    commit from @KIVagant, thanks!)
```
Comment (alpeb) [inline CHANGES.md]: That's opt-in, so I would be more precise and say something like "Added the `--wait-before-exit-seconds` flag to `linkerd inject` for the proxy sidecar to delay the start of its shutdown process". Not sure if we'd like to also add more context on the cases where adding that delay makes sense.

File: CHANGES.md [hunk 4]
```diff
@@ -1,3 +1,26 @@
...
+  * Added a pre-sign check to te identity service
+* Web UI
+  * Added logic to pause dashboard polling activity when the page is not in 
+    focus
+* Proxy
+  * Added a timeout to release resolutions to idle balancers
+  * Reclassified gRPC status codes as non-errors
```
Comment (olix0r) [inline CHANGES.md]: Maybe _Improved error classification for gRPC services_?

File: CHANGES.md [hunk 5]
```diff
@@ -1,3 +1,26 @@
...
+* Internal
+  * Updated helm charts to follow best practices using proper casing (thanks 
+    @Pothulapati!)
```
Comment (alpeb) [inline CHANGES.md]: This is an important BC breaking change that people should be aware of, so it can go under its own section `Helm` like we did back in the `edge-19.10.3` notes. Also `Helm` should be capitalized ;-)

File: CHANGES.md (hunk 1 above)
Comment (alpeb) [reply CHANGES.md]: Maybe also worth adding a note about the performance improvements for concurrent dashboard usage? (#3769)

File: CHANGES.md [hunk 6]
```diff
@@ -1,5 +1,23 @@
 ## edge-19.12.3
 
+This edge release adds support for pod IP and service cluster IP lookups, 
+improves performance of the dashboard, and makes `linkerd check --pre` perform 
+more comprehensive checks.
+
+The `--wait-before-seconds` flag has been added to allow Linkerd users to opt in
```
Comment (alpeb) [inline CHANGES.md]: should be `--wait-before-exit-seconds`
